File: ascending-data/BubbleSort/BubbleSort.php
<?php
function bubbleSort($arr) {
    $n = count($arr);
    for ($i = 0; $i < $n - 1; $i++) {
        for ($j = 0; $j < $n - $i - 1; $j++) {
            if ($arr[$j] > $arr[$j + 1]) {
                $temp = $arr[$j];
                $arr[$j] = $arr[$j + 1];
                $arr[$j + 1] = $temp;
            }
        }
    }
    return $arr;
}

$data = array_map('intval', file('ascending_data.txt', FILE_IGNORE_NEW_LINES | FILE_SKIP_EMPTY_LINES));
$start = microtime(true);
$sortedData = bubbleSort($data);
$end = microtime(true);
echo "Bubble Sort Execution Time: " . ($end - $start) . " seconds\n";

File: ascending-data/BubbleSort/BubbleSort.py
import time
def bubble_sort(arr):
    n = len(arr)
    for i in range(n - 1):
        for j in range(n - i - 1):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
def main():
    try:
        with open("ascending_data.txt", "r") as file:
            arr = list(map(int, file.read().split()))
    except FileNotFoundError:
        print("Error: ascending_data.txt file not found!")
        return
    start_time = time.time()
    bubble_sort(arr)
    end_time = time.time()
    execution_time = end_time - start_time
    print(f"Bubble Sort Execution Time: {execution_time * 1000:.4f} milliseconds")
if __name__ == "__main__":
    main()